### PR TITLE
Skip image compression when not desired (iOS)

### PR DESCRIPTION
**Pull request by fubar**
What do you guys think about this?

As reported in https://github.com/ivpusic/react-native-image-crop-picker/issues/252 and https://github.com/ivpusic/react-native-image-crop-picker/issues/449, file sizes end up around 3x larger when not specifying compression and resizing. Plus I need the ability to retrieve photos without any modifications.

This change skips the image compressor when the requested compression is 1 (or not specified) and max width/height are equal/larger than the original dimensions (or not specified). iOS only.

Feedback appreciated.

### CODE REVIEW

File: ios/ImageCropPicker.m [hunk 1]
```diff
@@ -429,6 +429,25 @@ - (NSDictionary*) createAttachmentResponse:(NSString*)filePath withExif:(NSDicti
              };
 }
 
+// See https://stackoverflow.com/questions/4147311/finding-image-type-from-nsdata-or-uiimage
+- (NSString *)determineMimeTypeFromImageData:(NSData *)data {
```
Comment (fubar) [inline ios/ImageCropPicker.m]: I am aware of this comment: https://github.com/ivpusic/react-native-image-crop-picker/pull/389/files#r129528647

I'd still suggest this solution for performance reasons as it seems to be very reliable mechanism.

Comment (fazeelzama20) [reply ios/ImageCropPicker.m]: @fubar Hello, forgive if I am asking silly question here, if we have to implement this, what changes we have to do in our code. I have implemented the library and using single image capture as well as multiple image capture functionality and it is working great. Can you tell me the procedure to skip the image compression in iOS.

**Comment by fubar** [comment]
@ivpusic Got a moment to look at this?

**Comment by ivpusic** [comment]
thanks!

**Comment by fubar** [comment]
Thanks for the merge! I have another PR in the pipes that adds an option to skip writing temporary files, just a heads up.

File: ios/ImageCropPicker.m (hunk 1 above)
Comment (fubar) [reply ios/ImageCropPicker.m]: This will be available in the next release.

**Comment by gianpaj** [comment]
Excuse my ignorance with ObjectC but what's the prop that can be set in JS component to disable image compression?

**Comment by superandrew213** [comment]
@gianpaj just don't set the compress fields.

**Comment by ryansolid** [comment]
@ivpusic Where did this land?

Is that still true? As far as I can tell if you don't set the compress fields doesn't it default to 0.8? Or the documentation atleast suggests as much. Is it possible to tell iOS not to compress the image file?

**Pull request merged**